packages/block-editor/src/components/block-styles/index.js

```diff
@@ -75,106 +77,114 @@ function BlockStyles( {
 
 	const { updateBlockAttributes } = useDispatch( blockEditorStore );
 	const genericPreviewBlock = useGenericPreviewBlock( block, type );
+	const [ hoveredStyle, setHoveredStyle ] = useState( null );
+	const debouncedSetHoveredStyle = useDebounce( setHoveredStyle, 250 );
+
+	const onStyleHover = useCallback(
```

Comment:
I don't think memoization is useful here based on how we use this callback below:

```js
onMouseEnter={ () => onStyleHover( style ) }
onFocus={ () => onStyleHover( style ) }
onMouseLeave={ () => onStyleHover( null ) }
onBlur={ () => onStyleHover( null ) }
```

Reply:
Good pick up, cheers. Removed. 👍

Reply:
Thank you ❤️